packages/web/src/components/track/OwnerActionButtons.tsx

```diff
@@ -0,0 +1,140 @@
+import { ReactNode, useCallback } from 'react'
+
+import { useGetPlaylistById, useGetTrackById } from '@audius/common/api'
+import { ID } from '@audius/common/models'
+import {
+  publishTrackConfirmationModalUIActions,
+  trackPageActions,
+  useEditTrackModal
+} from '@audius/common/store'
+import {
+  Flex,
+  IconButton,
+  IconPencil,
+  IconRocket,
+  IconShare
+} from '@audius/harmony'
+import { useDispatch } from 'react-redux'
+
+import Tooltip from 'components/tooltip/Tooltip'
+
+const { requestOpen: openPublishTrackConfirmationModal } =
+  publishTrackConfirmationModalUIActions
+const { makeTrackPublic } = trackPageActions
+
+const messages = {
+  share: 'Share',
+  edit: 'Edit Track',
+  publish: 'Make Public'
+}
+
+type OwnerActionButtonProps = {
+  contentId: ID
+  contentType: 'track' | 'collection'
+  isDisabled?: boolean
+  isLoading?: boolean
+  rightActions?: ReactNode
+  bottomBar?: ReactNode
+  isDarkMode?: boolean
+  isMatrixMode: boolean
+  showIconButtons?: boolean
+  onClickShare: (e?: any) => void
```

Comment:
nit: better type for this?

Reply:
good catch! I forgot about these `any` placeholders